cmd/metal-api/internal/datastore/machine.go

```diff
@@ -443,25 +445,69 @@ func (rs *RethinkStore) FindWaitingMachine(partitionid, sizeid string) (*metal.M
 			"value": string(metal.AvailableState),
 		},
 		"waiting": true,
-	}).Sample(1)
+	})
 
-	var available metal.Machines
-	err := rs.searchEntities(&q, &available)
+	var candidates metal.Machines
+	err := rs.searchEntities(&q, &candidates)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(available) < 1 {
+	var available metal.Machines
+
+	for _, m := range candidates {
+		liveliness, err := rs.EvaluateMachineLiveliness(m)
```

Comment:
We have a cronjob that evaluates the machine liveliness regularly, I think we do not really need to do that again. Just adding the liveliness to the filter query should be enough and safe us 20 lines of new code, right?